### PR TITLE
NRM-261 Add “Do you have information about where the exploiters are?”…

### PR DESCRIPTION
**Pull request by PaolaDMadd-Pro**
## What?
add new pages for exploiters location details with tests

## Why?
As requested by product owner per ticket [NRM 261](https://collaboration.homeoffice.gov.uk/jira/browse/NRM-261)

## How?
- add relevant views, validations, fields
- update submission and summary report files to include the new changes.
- add ui-Integration and acceptance tests for new pages
- update .snyk vulnerabilities file
- increased default timeout from 5 to 20min for acceptance test
- update puppeteer

## Testing?
unit test and integration test pass
acceptance tests: additional tests made exceeded timeout limit increased it to 20min.

## Check list
- [x] I have reviewed my own pull request for linting issues (e.g. adding new lines)
- [x] I have written tests (if relevant)
- [x] I have created a JIRA number for my branch
- [x] I have created a JIRA number for my commit
- [x] Ensure drone builds are green especially tests
- [x] I will squash the commits before merging

### CODE REVIEW

File: test/_ui-integration/nrm/validations.spec.js [hunk 1]
```diff
@@ -486,6 +486,55 @@ describe('validation checks of the nrm journey', () => {
     });
   });
 
+
```
Comment (Rhodine-orleans-lindsay) [inline test/_ui-integration/nrm/validations.spec.js:489]: remove extra line

File: test/_ui-integration/nrm/automatic-referral-application.spec.js [hunk 2]
```diff
@@ -272,13 +272,44 @@ describe('the journey of a nrm automatic referral application', () => {
...
+  it('goes to the information-exploiters-location page when user enters who-exploited-pv', async () => {
     const URI = '/who-exploited-pv';
     await initSession(URI);
     const response = await passStep(URI, {
       'who-exploited-pv': 'Ronald Testman'
     });
 
+    expect(response.text).to.contain('Found. Redirecting to /nrm/information-exploiters-location');
+  });
+
```
Comment (Rhodine-orleans-lindsay) [inline test/_ui-integration/nrm/automatic-referral-application.spec.js:284]: remove extra line

File: test/_ui-integration/nrm/application.spec.js [hunk 3]
```diff
@@ -256,14 +256,43 @@ describe('the journey of a nrm application', () => {
...
+  it('goes to the information-exploiters-location page when user enters who-exploited-pv', async () => {
     const URI = '/who-exploited-pv';
     await initSession(URI);
     const response = await passStep(URI, {
       'who-exploited-pv': 'Ronald Testman'
     });
 
+    expect(response.text).to.contain('Found. Redirecting to /nrm/information-exploiters-location');
+  });
+
```
Comment (Rhodine-orleans-lindsay) [inline test/_ui-integration/nrm/application.spec.js:268]: remove extra line

File: test/_ui-integration/nrm/application.spec.js [hunk 4]
```diff
@@ -256,14 +256,43 @@ describe('the journey of a nrm application', () => {
...
+  it('goes to the where-are-exploiters page when user enter are-exploiters-in-the-uk', async () => {
+    const URI = '/exploiters-uk';
+    await initSession(URI);
+    const response = await passStep(URI, {
+      'are-exploiters-in-the-uk': 'yes'
+    });
+
+    expect(response.text).to.contain('Found. Redirecting to /nrm/where-are-exploiters');
+  });
```
Comment (Rhodine-orleans-lindsay) [inline test/_ui-integration/nrm/application.spec.js:288]: add a new line after

File: apps/nrm/views/where-are-exploiters.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+
+
+{{<partials-page}}
+  {{$page-content}}
+    {{#renderField}}exploiters-current-location-details{{/renderField}}
+    {{#input-submit}}continue{{/input-submit}} {{> partials-save-and-exit-link}}
+  {{/page-content}}
+{{/partials-page}}
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/views/where-are-exploiters.html:8]: needs a new line after

File: apps/nrm/views/information-exploiters-location.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+
+
+{{<partials-page}}
+  {{$page-content}}
+    {{#renderField}}exploiters-location{{/renderField}}
+    {{#input-submit}}continue{{/input-submit}} {{> partials-save-and-exit-link}}
+  {{/page-content}}
+{{/partials-page}}
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/views/information-exploiters-location.html:8]: needs a  new line after

File: apps/nrm/views/information-exploiters-location.html [hunk 7]
```diff
@@ -0,0 +1,8 @@
+
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/views/information-exploiters-location.html:2]: remove extra lines

File: apps/nrm/views/exploiters-uk.html [hunk 8]
```diff
@@ -0,0 +1,6 @@
+{{<partials-page}}
+  {{$page-content}}
+    {{#renderField}}are-exploiters-in-the-uk{{/renderField}}
+    {{#input-submit}}continue{{/input-submit}} {{> partials-save-and-exit-link}}
+  {{/page-content}}
+{{/partials-page}}
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/views/exploiters-uk.html:6]: needs a new line after

File: apps/nrm/translations/src/en/pages.json [hunk 9]
```diff
@@ -325,6 +340,18 @@
       "who-exploited-pv": {
         "label": "Details of exploiter(s)"
       },
+
+      "exploiters-location": {
+        "label": "Do you have information about where the exploiters are?"
+      },
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/pages.json:347]: remove extra line

File: apps/nrm/translations/src/en/fields.json [hunk 10]
```diff
@@ -381,6 +381,36 @@
...
+    "options": {
+      "yes": {
+        "label": "Yes"
+      },
+      "no": {
+        "label": "No"
+      }
+    }
+  },
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:408]: remove extra line

File: apps/nrm/translations/src/en/fields.json [hunk 11]
```diff
@@ -381,6 +381,36 @@
...
+  "exploiters-current-location-details": {
+    "label": "What information do you have about where the exploiters are now?",
+    "hint": "This can be a general or specific location, or even a contact from another organisation who has this information."
+  },
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:413]: remove extra line

File: apps/nrm/translations/src/en/fields.json [hunk 12]
```diff
@@ -381,6 +381,36 @@
     "label": "Details about the exploiter(s)",
     "hint": "Include the languages the exploiters spoke, their gang or nicknames and if they had identifying features such as tattoos."
   },
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:384]: remove extra line

File: apps/nrm/translations/src/en/fields.json [hunk 13]
```diff
@@ -381,6 +381,36 @@
...
+    "options": {
+      "yes": {
+        "label": "Yes"
+      },
+      "no": {
+        "label": "No"
+      }
+    }
+  },
+
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:396]: remove extra line

File: apps/nrm/index.js [hunk 14]
```diff
@@ -291,11 +291,36 @@ module.exports = {
...
+      }],
+      next: '/types-of-exploitation'
+    },
+
+    '/exploiters-uk': {
+      behaviours: saveFormSession,
+      fields: ['are-exploiters-in-the-uk'],
+      next: '/where-are-exploiters'
+    },
+    '/where-are-exploiters': {
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/index.js:321]: I'd make the step name the same as the field name (exploiters-current-location-details), it fits the question more

File: apps/nrm/index.js [hunk 15]
```diff
@@ -291,11 +291,36 @@ module.exports = {
...
+
     '/who-exploited-pv': {
       behaviours: [
         saveFormSession
       ],
       fields: ['who-exploited-pv'],
+      next: '/information-exploiters-location'
+    },
+
+    '/information-exploiters-location': {
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/index.js:303]: I'd make the step name the same as the field name (exploiters-location)

**Review by Rhodine-orleans-lindsay — CHANGES_REQUESTED** [review]
A few suggestions and some syntax issues to correct

File: test/_ui-integration/nrm/validations.spec.js (hunk 1 above)
Comment (Rhodine-orleans-lindsay) [reply test/_ui-integration/nrm/validations.spec.js:489]: there is still an extra line there

File: test/_ui-integration/nrm/automatic-referral-application.spec.js (hunk 2 above)
Comment (Rhodine-orleans-lindsay) [reply test/_ui-integration/nrm/automatic-referral-application.spec.js:284]: there's still an extra line there

File: test/_ui-integration/nrm/application.spec.js (hunk 3 above)
Comment (Rhodine-orleans-lindsay) [reply test/_ui-integration/nrm/application.spec.js:268]: there is still an extra line

File: apps/nrm/translations/src/en/fields.json [hunk 16]
```diff
@@ -381,6 +381,32 @@
     "label": "Details about the exploiter(s)",
     "hint": "Include the languages the exploiters spoke, their gang or nicknames and if they had identifying features such as tattoos."
   },
+  "exploiters-location" : {
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:384]: format colon spaces

File: apps/nrm/translations/src/en/fields.json [hunk 17]
```diff
@@ -381,6 +381,32 @@
...
+    "options": {
+      "yes": {
+        "label": "Yes"
+      },
+      "no": {
+        "label": "No"
+      }
+    }
+  },
+  "are-exploiters-in-the-uk" : {
```
Comment (Rhodine-orleans-lindsay) [inline apps/nrm/translations/src/en/fields.json:395]: format colon spaces

File: test/_ui-integration/nrm/validations.spec.js (hunk 1 above)
Comment (PaolaDMadd-Pro) [reply test/_ui-integration/nrm/validations.spec.js]: path name changed.

**Review by Rhodine-orleans-lindsay — APPROVED** [review]
lgtm

**Review by sulthan-ahmed — APPROVED** [review]
for some reason the hint is being captured with the visually-hidden so @PaolaDMadd-Pro explained that they are providing a workaround.  As @Rhodine-orleans-lindsay is working on updating HOF, this will fix this workaround later so will approve this

**Pull request closed without merging**